### PR TITLE
Allow configuration of IIS config file location

### PR DESCRIPTION
**Pull request by WiredUK**
Something I needed for my own environment where I have IIS in a shared configuration environment which means the location of the `applicationHost.config` file is in a different folder.

This PR allows user to specify the location of the file:

    <iisConfigurationFile path="c:\myconfig.config"/>

This feature was also mentioned in https://github.com/projectkudu/kudu/issues/1438

### CODE REVIEW

File: Kudu.SiteManagement/Configuration/IKuduConfiguration.cs [hunk 1]
```diff
@@ -18,6 +18,7 @@ public interface IKuduConfiguration
         string RootPath { get; }
         string ApplicationsPath { get; }
         string ServiceSitePath { get; }
+        string IisConfigurationFile { get; }
```
Comment (davidebbo) [inline Kudu.SiteManagement/Configuration/IKuduConfiguration.cs]: I'd capitalize IIS here (and other places). 'Iis' just doesn't look right :)

File: Kudu.Web/Web.config [hunk 2]
```diff
@@ -27,6 +27,15 @@
     <serviceSite path="..\Kudu.Services.Web"/>
     <applications path="..\apps"/>
     <!--
+    IIS Shared Configuration
+    ======================================================================================================================
+    
+    If your IIS config file is in a different location (usually due to running a shared configration, you can specify the
```
Comment (davidebbo) [inline Kudu.Web/Web.config]: Missing closed parenthesis in comment

File: Kudu.SiteManagement/Configuration/IKuduConfiguration.cs (hunk 1 above)
Comment (WiredUK) [inline Kudu.SiteManagement/Configuration/IKuduConfiguration.cs]: That's Resharper getting it's knickers in a twist. Shall I redo it?

File: Kudu.SiteManagement/Configuration/IKuduConfiguration.cs [hunk 3]
```diff
@@ -79,6 +80,21 @@ public string ApplicationsPath
...
 
+        private const string DefaultIisConfigurationFile = "%windir%\\system32\\inetsrv\\config\\applicationHost.config";
+
+        public string IisConfigurationFile
+        {
+            get
+            {
+                if(_section == null || _section.IisConfigurationFile == null)
+                    return Environment.ExpandEnvironmentVariables(DefaultIisConfigurationFile);
+
```
Comment (davidebbo) [inline Kudu.SiteManagement/Configuration/IKuduConfiguration.cs]: Style: we use braces even for single liners.

Comment (WiredUK) [inline Kudu.SiteManagement/Configuration/IKuduConfiguration.cs]: Slightly confused on this one as the rest of this file uses the same style (I tried to match it)

File: Kudu.SiteManagement/SiteManager.cs [hunk 4]
```diff
@@ -171,7 +171,7 @@ private static List<string> GetSiteUrls(IIS.Site site)
                 {
                     try
                     {
-                        DeleteSiteAsync(applicationName).Wait();
+                        await DeleteSiteAsync(applicationName);
```
Comment (davidebbo) [inline Kudu.SiteManagement/SiteManager.cs]: This looks unrelated, no?

File: Kudu.SiteManagement/Configuration/IKuduConfiguration.cs (hunk 3 above)
Comment (davidebbo) [inline Kudu.SiteManagement/Configuration/IKuduConfiguration.cs]: That's because they were contributed and we didn't catch it :)

File: Kudu.SiteManagement/Configuration/IKuduConfiguration.cs (hunk 1 above)
Comment (davidebbo) [inline Kudu.SiteManagement/Configuration/IKuduConfiguration.cs]: Yes, please do change it.

**Comment by davidebbo** [comment]
Change looks good, just various minor comments.

File: Kudu.SiteManagement/SiteManager.cs (hunk 4 above)
Comment (WiredUK) [inline Kudu.SiteManagement/SiteManager.cs]: Wasn't sure if it should go in as a separate PR? It got it's own commit though. Shall I reverse it?

Comment (davidebbo) [inline Kudu.SiteManagement/SiteManager.cs]: It's ok, then. I'll keep that commit separate, and squash all the others related to this PR.

File: Kudu.SiteManagement/Configuration/IKuduConfiguration.cs (hunk 3 above)
Comment (davidebbo) [inline Kudu.SiteManagement/Configuration/IKuduConfiguration.cs]: Don't worry about this, I'll fix them all in that file.

**Comment by davidebbo** [comment]
Rebased, squashed and pushed. Thanks for the contrib!

**Comment by WiredUK** [comment]
You are welcome, and thanks for an awesome project! Perhaps you want to close issue #1438 also?
